src/sentry/incidents/logic.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import absolute_import
+
+from django.db import transaction
+
+from sentry.incidents.models import (
+    Incident,
+    IncidentGroup,
+    IncidentProject,
+)
+
+
+def create_incident(
```

Comment:
Could/Should we use the Mediators here? cc @mnoble

Reply:
I think so. You get all the transaction stuff for "free", as well as some useful logging and type safety. I generally like having a place where I can go to look for all domain logic around resources. It could work into this pattern with each "app" just having a mediators directory. I like that pattern and naming, but this file effectively achieves the same thing, so I'm not _super_ concerned.

Reply:
Do we plan to use mediators everywhere going forward? For the moment I'd like to leave this as is since it has been sitting a while. If we want to start using mediators more we can refactor.

Reply:
>  If we want to start using mediators more I can refactor.

I think having two 'new' ways to handle logic outside of endpoints isn't great and would rather see us adopt only one way and gradually normalize on that approach for existing code.

Reply:
I agree in general, but I don't know that we've standardized on anything at this point. If we're looking to push a standard pattern everywhere then I think we need to meet and formalize it, and get buy-in from others in the company.